### PR TITLE
proposal for combined logging

### PR DESCRIPTION
**Pull request by wwitzel3**
Signed-off-by: Wayne Witzel III <wayne@riotousliving.com>

**What this PR does / why we need it**:
Proposal for updates to our logging.

**Special notes for your reviewer**:
Please leave feedback and suggestions.

### CODE REVIEW

**Comment by mklanjsek** [comment]
This looks great!  Additionally, we should try to add as much metadata about each log line as possible. Having that level of detail provides much richer logging experience and filtering capabilities.  This is specially true for labels as some of the best logging platforms are based on label tagging/filtering, for example [Loki](https://grafana.com/oss/loki/)

File: proposals/20200224-combined-logging.md [hunk 1]
```diff
@@ -0,0 +1,72 @@
...
+Support the ability to view all of the container logs for a given workload. Currently you must request each container
+log separately.
+
+## Goals
+ - Combine the logging output from multiple containers.
+ - Tag/mark which log lines are from which container.
+ - Provide a list of which container logs are included in the stream.
+ - Only send content when there are new log entries.
+
+## Non-Goals
```
Comment (bryanl) [inline proposals/20200224-combined-logging.md]: Another suggestion I want to offer is that is combined logging for a pod. Multiple pods are not in scope.

File: proposals/20200224-combined-logging.md [hunk 2]
```diff
@@ -0,0 +1,72 @@
...
+Add a sentinel value of `AllContainers` to Octant and make this the default choice when requesting logs.
+
+Add a set of new actions to Octant to handle starting a log stream.
+
+Add a new EventType to Octant for log streams that the WebsocketService can subscribe to.
+
+Extend our existing log entry to include the container the log entry is from.
+
+#### Stream Aggregation
+Create a go routine that wraps the `Stream` call for each containers Octant will stream logs for.
```
Comment (bryanl) [inline proposals/20200224-combined-logging.md]: Is this a good opportunity to create a better logging interface? I’m not 100% sure what it should look like but the ideas seems to be in the right direction.

File: proposals/20200224-combined-logging.md (hunk 1 above)
Comment (wwitzel3) [reply proposals/20200224-combined-logging.md]: I'm thinking about this some more. I'm going to consider what complexity is added if we say this is to get logs combined from any workload vs. for all containers of a pod. I'm also going to consider how we might do this for only pods, but in way that makes it extensible for the next improvement, which would Application logs.

File: proposals/20200224-combined-logging.md (hunk 2 above)
Comment (wwitzel3) [reply proposals/20200224-combined-logging.md]: Yeah, I think we will ultimately end up with a different interface. I will walk through some ideas and update the proposal to reflect the new design.

I'm thinking this change will create an `internal/logging` module which will house the action handlers, streaming functions, and helpers.

**Review by bryanl — COMMENTED** [review]
My thoughts after thinking about this are the goals specified need a strategy to go along with them under the proposal setting. How are you going to work towards each goal?

**Review by bryanl — APPROVED** [review]

**Review by bryanl — APPROVED** [review]

**Pull request merged**